test/ProgressOnderwijsUtils.Tests/HtmlTests.cs

```diff
@@ -40,23 +40,49 @@ public void IFrameGetsClosingTag()
         }
 
         [Fact]
-        public void AppendingToFragmentsWorks() { 
+        public void AppendingToFragmentsWorks()
+        {
             var html = _div.AsFragment().Append("test").SerializeToStringWithoutDoctype();
             PAssert.That(() => html == "<div></div>test");
         }
 
         [Fact]
-        public void AppendingThreeThingsWorks() { 
+        public void AppendingThreeThingsWorks()
+        {
             var html = _div.AsFragment().Append("test", _b.Content("la")).SerializeToStringWithoutDoctype();
             PAssert.That(() => html == "<div></div>test<b>la</b>");
         }
 
         [Fact]
-        public void PerfIterativeAppendingDoesNotCreateLotsOfArrays () { 
+        public void PerfIterativeAppendingDoesNotCreateLotsOfArrays()
+        {
             var html = "1".AsFragment().Append("2").Append("3").Append("4");
             var kids = (HtmlFragment[])html.Content;
             PAssert.That(() => kids.First().SerializeToStringWithoutDoctype() == "1");
             PAssert.That(() => kids.Last().SerializeToStringWithoutDoctype() == "4");
         }
+
+        [Fact]
+        public void AllInterestingPlusOperatorOrderingsCompileAndReturnHtmlFragment()
+        {
+            var a = _b + _p;
+            var b = "asd" + _q;
+            var c = _i + "asd";
+            var d = _a + new CustomHtmlElement("bla");
+            var e = new CustomHtmlElement("bla") + _s;
+            var f = new CustomHtmlElement("bla") + new CustomHtmlElement("bla");
+            var g = "test" + new CustomHtmlElement("bla");
+            var h = new CustomHtmlElement("bla") + "test";
+            var i = HtmlFragment.Empty + _q;
+            var j = _i + HtmlFragment.Empty;
+            var k = HtmlFragment.Empty + new CustomHtmlElement("bla");
+            var l = new CustomHtmlElement("bla") + HtmlFragment.Empty;
+            var m = HtmlFragment.Empty + HtmlFragment.Empty;
+            var n = "test" + HtmlFragment.Empty;
+            var o = HtmlFragment.Empty + "test";
+
+            var typeReflectionHack = new { a, b, c, d, e, f, g, h, i, j, k, l, m, n, o };
+            PAssert.That(() => typeReflectionHack.GetType().GetProperties().Select(fi => fi.PropertyType).Distinct().SequenceEqual(new[] { typeof(HtmlFragment) }));

```

Comment:
Why's this better than having separate tests?

Reply:
Seperate tests would mean lots and lots of copy-paste almost identical tests, which typically means people's eyes glaze over and they never notice is something is copied incorrectly.

Additionally, this is shorter, and the real test is in any case that this compiles, not so much the assertion (hence the fact name).

Reply:
(or do you mean copy-paste assertions?)